core/src/main/scala/kafka/cluster/Replica.scala

```diff
@@ -165,9 +168,16 @@ class Replica(val brokerId: Int,
       s"non-local replica $brokerId"))
   }
 
+  /*
+   * Convert hw to local offset metadata by reading the log at the hw offset.
+   * If the hw offset is out of range, return the first offset of the first log segment as the offset metadata.
+   */
   def convertHWToLocalOffsetMetadata() = {
```

Comment:
If you're adding another commit to address Ismael's comment, could you also change this method to have `Unit` return type? It's a little confusing otherwise.

Reply:
Good catch. It is updated now to follow the same style of other unit-typed method in the class.